### ISSUE TITLE
Steps to run NPM

### ISSUE THREAD

**Issue by shashankg7** [harrywy/NPM#1]
Hi,

Thanks for open sourcing NPM. I want to run the model on dataset mentioned in the paper. I have downloaded the sql dumps. Can you please tell me steps I should follow to reproduce the results? I have run the sql dumps through mysql and created the 3 tables. 

I am struggling to understand the instructions in the README file. It will be really helpful if you can biefly describe the steps I should follow from start. 

Thanks

**Comment by inzaghi250** [comment]
Hi Wenyi,

I am also feeling hard reproducing the experiments. 1) The codes in paper2vec.cpp do not look like what is described in the paper (or the lines that do have been commented out). 2) For indexing, is indexing-soft-bayes-log exactly the one in the paper?

Thanks.

**Comment by harrywy** [comment]
@inzaghi250 
1) paper2vec.cpp is for comparison basically it is the word2vec implementation, so the result will be w2v result in the paper. The main code, as indicated in the README file in src/ is citation-model-MT-inverse.cpp.

 2) I think it is the index-soft-bayes.cpp. The only difference is whether there is log there, so I kind of forget which one gives a better result. But if I remembered right it is the without log one.

**Comment by harrywy** [comment]
@shashankg7 I added the data preparation code. Hope it will help. It is used to get data out of database

**Comment by shashankg7** [comment]
@harrywy thanks Wenyi for your help. Appreciate it!

**Comment by inzaghi250** [comment]
@harrywy Thanks for the reply. I am using citation-model-MT-inverse now. Could you tell me how long it took you to train with default parameters (i.e., negative=1000, iter=1000, dim=600)? On my 10-core machine I need 24h to finish 10 iters, even if I use negative=100.

**Comment by harrywy** [comment]
@inzaghi250  I was using a 24-core (not 24 hyper-threads) server, and it took about a week to get to 200 iterations. If I recalled right, I use 200 iteration result for experiment.  So if you are using 10-core, 10 thread will probably be faster.

**Comment by inzaghi250** [comment]
@harrywy Thanks for the suggestion. I am asking a lot because it is an important baseline for my ongoing project. Sorry but there are still a few:

1. Is the learning rate alpha=1 in the codes used in the presented experiments?

2. How many documents did you index for each word (i.e., the parameter N in index-soft-bayes) Is the default 10 too few?

3. How did you discard non-adjs and non-nouns for training? By a POS tagger or by a stop-word list?

4. Did you attempted to compare with a simple BM25 baseline? I.e., each paper is given a virtual document containing all its citation contexts, and a testing context is queried on the virtual documents for potential cited papers.

Thanks!
Inzaghi250

**Comment by inzaghi250** [comment]
@harrywy Thanks for the suggestion. I am asking a lot because it is an important baseline for my ongoing project. Sorry but there are still a few:

1.Is the learning rate alpha=1 in the codes used in the presented experiments?

2.How many documents did you index for each word (i.e., the parameter N in index-soft-bayes) Is the default 10 too few?

3.Did you attempted to compare with a simple BM25 baseline? I.e., each paper is given a virtual document containing all its citation contexts, and a testing context is queried on the virtual documents for potential cited papers. Did you attempted to compare with Qi He's WWW'10 method?


Thanks!
 Inzaghi250

**Comment by harrywy** [comment]
1. Yes.
2. I used 10 for evaluation.
3. I didn't compare with BM25.  Nor did I compare with Qi He's method, we don't have the code and the method is not scalable on such large dataset for testing phase.

**Comment by xanhho** [comment]
Hi Wenyi,

I also feeling hard reproducing the experiments.
Can you please describe the steps I should follow from start?

Thank you very much!
Xanh Ho
